client/extensions/relatedArticles/src/EventsRelatedArticles/EventsRelatedArticlesModal.tsx

```diff
@@ -0,0 +1,307 @@
+            method: "GET",
+            path: "/search_providers",
+            urlParams: {
+                manage: 1,
+            }
+        }).then((result) => {
+            const repoId = result._items.find((provider) => provider.source === 'http://wss-01.staging.belga.be:9000/archivenewsobjects')._id;
+
+            this.setState({
+                repo: repoId,
+            })
+        })
+    }
+
+    componentDidUpdate(_prevProps: Readonly<IProps>, prevState: Readonly<IState>): void {
+        if (prevState.activeLanguage.code !== this.state.activeLanguage.code
+            || prevState.searchQuery !== this.state.searchQuery
+            || prevState.repo !== this.state.repo
+        ) {
+            this.setState({
+                loading: true,
+            })
+        }
+    }
+
+    render(): React.ReactNode {
+        const {closeModal} = this.props;
+        const allLanguages = [
+            {
+                label: 'English',
+                code: 'en'
+            },
+            {
+                label: 'Dutch',
+                code: 'nl'
+            },
+            {
+                label: 'French',
+                code: 'fr'
+            },
+            {
+                label: 'All languages',
+                code: ''
+            }
+        ];
+        const onSelectLanguage = (lang: typeof allLanguages[0]) => {
+            this.setState({
+                activeLanguage: lang
+            })
+        }
+
+        return (
+            <Modal
+                headerTemplate={gettext('Search Related Articles')}
+                visible
+                contentPadding="none"
+                contentBg="medium"
+                size="x-large"
+                onHide={closeModal}
+                footerTemplate={(
+                    <Spacer h gap="8" noWrap alignItems="end" justifyContent="end">
+                        <Button
+                            onClick={() => {
+                                closeModal();
+                            }}
+                            text={gettext('Cancel')}
+                            style="filled"
+                        />
+                        <Button
+                            onClick={() => {
+                                this.props.onChange(cleanArticlesFields(this.state.currentlySelectedArticles ?? []))
+                                closeModal();
+                            }}
+                            disabled={JSON.stringify(this.props.selectedArticles) === JSON.stringify(this.state.currentlySelectedArticles)}
+                            text={gettext('Apply')}
+                            style="filled"
+                            type="primary"
+                        />
+                    </Spacer>
+                )}
+            >
+                <Spacer v gap="0">
+                    <div
+                        style={{
+                            padding: 12,
+                            backgroundColor: 'white'
```

Comment:
Does this work with a dark theme? An issue with using inline styles vs styles defined in the UI-Framework